github/src/main/java/com/redhat/ipaas/github/GitHubService.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.redhat.ipaas.github;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.eclipse.egit.github.core.Repository;
+import org.eclipse.egit.github.core.client.GitHubClient;
+import org.eclipse.egit.github.core.service.RepositoryService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class GitHubService {
+
+    private final RepositoryService repoService;
+
+    public GitHubService() {
+        // Maybe make the service name configurable ?
+        GitHubClient client = new GitHubClient("ipaas-github-proxy");
```

Comment:
just hostname, according to the example at https://github.com/eclipse/egit-github/tree/master/org.eclipse.egit.github.core#using-github-enterprise

Still need to make a full turnaround ;-)